magma/backend/coreir_.py

```diff
@@ -81,445 +32,55 @@ def get_instance(self):
         return self.__instance
 
     def reset_instance(self):
-        self.__instance = coreir.Context()
+        self.__instance = Context()
 
     def __init__(self):
-        self.__instance = coreir.Context()
+        self.__instance = Context()
+
+
 CoreIRContextSingleton()
 
+_context_to_modules = {}
+
+
 class CoreIRBackend:
-    context_to_modules_map = {}
-    def __init__(self, context=None, check_context_is_default=True):
```

Comment:
IIRC removing `check_context_is_default=True` will break aetherling code since there's some code that creates these singleton contexts and uses this flag to override the warning (disabling it).  Perhaps we can add this back just for backwards compatibility?

Reply:
My old tests used to create coreir contexts. My new tests don't they just need access to `CoreIRContextSingleton().reset_instance()`. See https://github.com/David-Durst/aetherling/blob/master/conftest.py

Reply:
Got it. In either case it was just a warning (so unless you were checking the output for a warning, it shouldn't affect the result of tests). @David-Durst confirming this change is ok?